### PR TITLE
BugFix: loading to external olap table causes be crash

### LINKED ISSUE #4573: Load to olap external table causes be crash.
### Steps to reproduce the behavior (Required)
1. CREATE EXTERNAL OLAP Table and the host is non-master fe' host.
2. load to the external table.

### Expected behavior (Required)
load successfully.

### Real behavior (Required)
be crashed with FATAL log:
F0328 10:26:50.170414  8962 txn_manager.cpp:158] invalid commit req  partition_id=991573 transaction_id=0 tablet_id=992498

### StarRocks version (Required)
 - 2.0.3

### PR DESCRIPTION
**Pull request by gengjun-git**
## What type of PR is this：
- [x] bug
- [ ] feature
- [ ] enhancement
- [ ] others

## Which issues of this PR fixes ：

Fixes #4573

## Problem Summary(Required) ：

Throws exception when response status is not OK.

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 0 / 38 (00.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/master/MasterImpl.java|0|3|00.00%|
|:large_blue_circle:|com/starrocks/transaction/GlobalTransactionMgr.java|0|35|00.00%|

File: fe/fe-core/src/main/java/com/starrocks/master/MasterImpl.java [hunk 1]
```diff
@@ -1201,7 +1201,7 @@ public TBeginRemoteTxnResponse beginRemoteTxn(TBeginRemoteTxnRequest request) th
                     new TxnCoordinator(TxnSourceType.FE, FrontendOptions.getLocalHostAddress()),
                     LoadJobSourceType.valueOf(request.getSource_type()), request.getTimeout_second());
         } catch (Exception e) {
-            LOG.info("begin remote txn error, label {}, msg {}", request.getLabel(), e.getStackTrace());
+            LOG.warn("begin remote txn failed, label {}", request.getLabel(), e);
```
Comment (ABingHuang) [inline fe/fe-core/src/main/java/com/starrocks/master/MasterImpl.java:1204]: add warning log in line 1191

File: fe/fe-core/src/main/java/com/starrocks/transaction/GlobalTransactionMgr.java [hunk 2]
```diff
@@ -144,94 +135,93 @@ public long beginRemoteTransaction(long dbId, List<Long> tableIds, String label,
...
+            String errStr;
+            if (response.status.getError_msgs() != null) {
+                errStr = String.join(",", response.status.getError_msgs());
+            } else {
+                errStr = "";
+            }
+            LOG.warn("call fe {} beginRemoteTransaction rpc method failed, label: {}, error: {}", addr, label, errStr);
+            throw new BeginTransactionException(errStr);
+        } else {
+            LOG.info("begin remote txn, label: {}, txn id: {}", label, response.getTxn_id());
```
Comment (ABingHuang) [inline fe/fe-core/src/main/java/com/starrocks/transaction/GlobalTransactionMgr.java:155]: suggest to add a check, if response.getTxn_id() <= 0, throw an exception.

File: fe/fe-core/src/main/java/com/starrocks/transaction/GlobalTransactionMgr.java [hunk 3]
```diff
@@ -144,94 +135,93 @@ public long beginRemoteTransaction(long dbId, List<Long> tableIds, String label,
...
         }
-        ClientPool.frontendPool.returnObject(addr, client);
-        return response.getTxn_id();
+        if (response.status.getStatus_code() != TStatusCode.OK) {
+            String errStr;
+            if (response.status.getError_msgs() != null) {
+                errStr = String.join(",", response.status.getError_msgs());
+            } else {
+                errStr = "";
+            }
```
Comment (ABingHuang) [inline fe/fe-core/src/main/java/com/starrocks/transaction/GlobalTransactionMgr.java:151]:
```suggestion
            String errStr = "";
            if (response.status.getError_msgs() != null) {
                errStr = String.join(",", response.status.getError_msgs());
            }
```

File: fe/fe-core/src/main/java/com/starrocks/transaction/GlobalTransactionMgr.java [hunk 4]
```diff
@@ -144,94 +135,93 @@ public long beginRemoteTransaction(long dbId, List<Long> tableIds, String label,
...
+            LOG.warn("call fe {} commitRemoteTransaction rpc method failed, txn id: {}, error: {}", addr, transactionId, errStr);
+            throw new TransactionCommitFailedException(errStr);
+        } else {
+            LOG.info("commit remote, txn id: {}", transactionId);
+            return true;
+        }
     }
 
     // abort transaction in remote DorisDB cluster
```
Comment (wuleistarrocks) [inline fe/fe-core/src/main/java/com/starrocks/transaction/GlobalTransactionMgr.java:194]: change to StarRocks :)

**Review by wuleistarrocks — DISMISSED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 0 / 48 (00.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/master/MasterImpl.java|0|9|00.00%|
|:large_blue_circle:|com/starrocks/transaction/GlobalTransactionMgr.java|0|39|00.00%|

**Review by chaoyli — APPROVED** [review]

**Review by Astralidea — APPROVED** [review]

**Pull request merged**